elastic/kibana
# Onboard Uptime rule types with FAAD

Towards: https://github.com/elastic/kibana/issues/169867

This PR onboards Uptime rule types (Tls, Duration Anolamy and Monitor status) with FAAD.

We are deprecating the rule-registry plugin and onboard the rule types with the new alertsClient to manage alert-as-data.
There is no new future, all the rule types should work as they were, and save alerts with all the existing fields. 

## To verify:

- Switch to Kibana 8.9.0 in your local repo. (In this version Uptime rules are not deprecated)
- Run your ES with: `yarn es snapshot -E path.data=../local-es-data`
- Run your Kibana
- Create Uptime rules with an active and a recovered action (You can run Heartbeat locally if needed, [follow the instructions](https://www.elastic.co/guide/en/beats/heartbeat/current/heartbeat-installation-configuration.html))
- Stop your ES and Kibana
- Switch to this branch and run your ES with  `yarn es snapshot -E path.data=../local-es-data` again.
- Run your Kibana
- Modify Uptime rulesType codes to force them to create an alert.
Example:
Mock [availabilityResults in status_check](https://github.com/elastic/kibana/blob/main/x-pack/plugins/observability_solution/uptime/server/legacy_uptime/lib/alerts/status_check.ts#L491) with below data
```
availabilityResults = [
      {
        monitorId: '1',
        up: 1,
        down: 0,
        location: 'location',
        availabilityRatio: 0.5,
        monitorInfo: {
          timestamp: '',
          monitor: {
            id: '1',
            status: 'down',
            type: 'type',
            check_group: 'default',
          },
          docId: 'docid',
        },
      },
    ];
```

It should create an alert. The alert should be saved under `.alerts-observability.uptime.alerts` index and be visible under observability alerts page.

Then remove the mock, the alert should be recovered.
